src/utility/resource_lock.cpp

```diff
@@ -0,0 +1,80 @@
+/**
+ * Copyright (c) 2011-2015 libbitcoin developers (see AUTHORS)
+ *
+ * This file is part of libbitcoin.
+ *
+ * libbitcoin is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License with
+ * additional permissions to the one published by the Free Software
+ * Foundation, either version 3 of the License, or (at your option)
+ * any later version. For more information see LICENSE.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+#include <bitcoin/bitcoin/utility/resource_lock.hpp>
+
+#include <boost/filesystem.hpp>
+#include <boost/interprocess/exceptions.hpp>
+#include <bitcoin/bitcoin/unicode/ifstream.hpp>
+#include <bitcoin/bitcoin/unicode/ofstream.hpp>
+#include <bitcoin/bitcoin/utility/assert.hpp>
+
+namespace libbitcoin {
+
+void touch_file(const file_path& path)
```

Comment:
Prefer private static method to file scope function. This facilitates testing of the method (in which case it would need to become protected) and more strongly organizes code.